dom.go

```diff
@@ -418,20 +454,26 @@ func renderComponent(comp Component, prevRender *HTML) (h *HTML, skip bool) {
 		nextRender = Tag("noscript")
 	}
 
-	// Restore the actual rendered HTML.
+	// Reconcile the actual rendered HTML.
 	nextRender.reconcile(prevRender)
 
 	// Update the context to consider this render.
-	comp.Context().prevRender = nextRender
-	comp.Context().prevComponent = comp
-	comp.Context().prevRenderComponent = doCopy(comp)
+	//
+	// TODO: In the event that this really is the previous component instance,
```

Comment:
Ok, fixing the above actually turned out to be quite easy. I just needed to make the persistent component instance always be the source of truth for Context information.

I've done this in https://github.com/gopherjs/vecty/pull/130/commits/df133f3e38d8d450bd1567b4c77e65f8d9d1f09f and it fixes that TODO (see e391b1bb41223afa8b5cf322f58f269bc65e64d2)